src/main/java/backend/resource/TurboUser.java

```diff
@@ -90,9 +92,9 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         TurboUser that = (TurboUser) o;
-        return loginName.equals(that.loginName) &&
-                realName.equals(that.realName) &&
-                avatarURL.equals(that.avatarURL);
+        return (loginName == that.loginName || (loginName != null && loginName.equals(that.loginName))) &&
```

Comment:
Could this be fixed by moving the null checks to the `User` constructor instead (and defaulting to `""`)? I was hoping to avoid `null` entirely in the model classes.